src/ui/pages/Backup/VerifyBackup.tsx

```diff
@@ -31,23 +55,35 @@ export function VerifyBackup({ onSuccess }: { onSuccess(): void }) {
     type: isTechnicalHint ? 'text' : undefined,
   });
 
-  const { data: mnemonic, isLoading, error } = usePendingRecoveryPhrase();
+  const backupContext = useBackupContext();
+  const {
+    data: recoveryPhrase,
+    isLoading,
+    error,
+  } = useRecoveryPhrase(backupContext);
 
   useEffect(() => {
     if (isSessionExpiredError(error)) {
-      navigate('/onboarding/session-expired', { replace: true });
+      if (backupContext.appMode === 'onboarding') {
```

Comment:
Looks like this logic should be described inside external param onSessionExpired
In general, it would be nice to guard backup flow from describing different behaviour for different context as much as possible and move everything to the top level.

Reply:
Agree, we can improve this even more!
Let me try to extract this session-expiration-checking logic and move it to the parent component